packages/runtime/src/helpers/files.ts

```diff
@@ -334,7 +334,13 @@ const getServerFile = (root: string, includeBase = true) => {
 }
 
 const baseServerReplacements: Array<[string, string]> = [
-  [`let ssgCacheKey = `, `let ssgCacheKey = process.env._BYPASS_SSG || `],
+  // force manual revalidate during cache fetches
+  [
+    `checkIsManualRevalidate(req, this.renderOpts.previewProps)`,
+    `checkIsManualRevalidate(process.env._REVALIDATE_SSG ? { headers: { 'x-prerender-revalidate': this.renderOpts.previewProps.previewModeId } } : req, this.renderOpts.previewProps)`,
```

Comment:
Do we know if `process.env._REVALIDATE_SSG` could ever be 'false' (which would result in an unintentional truthy evaluation)?

I'm also not certain about if `previewProps` are required within `renderOpts`, but just in case - is there a chance that it could be `undefined` such that an error would be thrown when `previewModeId` is accessed?

Reply:
Nope, `_REVALIDATE_SSG` will never be false when deployed on Netlify. It will be undefined if the code runs outside of our handler functions, e.g. locally with `next dev` (unless a user explicitly set the env var, but it's not publicly documented).

Thanks for the heads-up, but I've double-checked the Next server code and `renderOpts` will always contain `previewProps`, which will always contain `previewModeId`.